src/map.cpp

```diff
@@ -5695,18 +5695,26 @@ const visibility_variables &map::get_visibility_variables_cache() const {
 
 lit_level map::apparent_light_at( const tripoint &p, const visibility_variables &cache ) const {
     const int dist = rl_dist( g->u.pos(), p );
+    
+    // Being outside of the world bubble overrides Clairvoyance.
```

Comment:
This shouldn't be necessary, wasn't the fix for this in handle_mouseview?

Reply:
Finally getting back to this PR.  So apparently this function isn't only called from handle_mouseview, which is why I ended up handling it here instead of in handle_mouseview.

Reply:
It is used heavily in `map::update_visibility_cache`, which is called on the entire bubble and only bubble (ie. it doesn't ever cross boundaries of the bubble).
`update_visibility_cache` is not cheap and `apparent_light_at` is the most expensive part of it.

Reply:
`map::inbounds` isn't an expensive check though.  I'm not sure how else to do it.

Reply:
Extract the check to `print_all_tile_info`, which is the only place where this check matters.

Reply:
Ah, I see.  Done.